pkg/neonvm/controllers/vm_controller.go

```diff
@@ -239,20 +235,15 @@ func (r *VMReconciler) doFinalizerOperationsForVirtualMachine(ctx context.Contex
 	log := log.FromContext(ctx)
 
 	// The following implementation will raise an event
-	r.Recorder.Event(vm, "Warning", "Deleting",
-		fmt.Sprintf("Custom Resource %s is being deleted from the namespace %s",
-			vm.Name,
-			vm.Namespace))
+	log.Info("Deleting VirtualMachine")
 
 	// Release overlay IP address
 	if vm.Spec.ExtraNetwork != nil {
 		ip, err := r.IPAM.ReleaseIP(ctx, types.NamespacedName{Name: vm.Name, Namespace: vm.Namespace})
 		if err != nil {
-			return fmt.Errorf("fail to release IP: %w", err)
+			return fmt.Errorf("fail to release overlay IP: %w", err)
 		}
-		message := fmt.Sprintf("Released IP %s", ip.String())
-		log.Info(message)
-		r.Recorder.Event(vm, "Normal", "OverlayNet", message)
+		log.Info(fmt.Sprintf("Released overlay IP %s", ip.String()))
```

Comment:
note: the idiomatic way to log with this lib is probably more like:
```suggestion
		log.Info("Released overlay IP", ip, ip.String())
```

Reply:
https://pkg.go.dev/github.com/go-logr/logr#Logger.Info has more examples

Reply:
+1; see "notes for review" in the PR description:

> I think there's also a lot of cleanup we should to make our log lines more structured, but better to leave that for a follow-up IMO.